src/main/java/org/vafer/jdeb/DataBuilder.java

```diff
@@ -272,12 +273,20 @@ private void createParentDirectories( String filename, String user, long uid, St
             }
         };
 
+        boolean finishedWithoutErrors = true;
         try {
             for (DataProducer data : producers) {
                 data.produce(receiver);
             }
+        } catch(Exception e) {
+            finishedWithoutErrors = false;
+            throw new RuntimeException("Couldn't process all data producers", e);
```

Comment:
you want the RuntimeException so that fails the build IIUC?

Reply:
I was thinking about using using `PackagingException` but that would require adding to method signature. Could that be potential breaking change?

Reply:
The most interesting context is the caught exception.

Reply:
ah, learning a bit how exception handling :-D  changed to just rethrow original exception.